File: Cxx/PolyData/ClosedSurface.cxx
#include <vtkSmartPointer.h>
#include <vtkSphereSource.h>
#include <vtkFeatureEdges.h>

int main(int, char *[])
{
  // Create a sphere
  vtkSmartPointer<vtkSphereSource> sphereSource =
    vtkSmartPointer<vtkSphereSource>::New();
  sphereSource->Update();

  vtkSmartPointer<vtkFeatureEdges> featureEdges =
    vtkSmartPointer<vtkFeatureEdges>::New();
  featureEdges->FeatureEdgesOff();
  featureEdges->BoundaryEdgesOn();
  featureEdges->NonManifoldEdgesOn();
  featureEdges->SetInputConnection(sphereSource->GetOutputPort());
  featureEdges->Update();

  int numberOfOpenEdges = featureEdges->GetOutput()->GetNumberOfCells();

  if(numberOfOpenEdges > 0)
  {
    std::cout << "Surface is not closed" << std::endl;
  }
  else
  {
    std::cout << "Surface is closed" << std::endl;
  }

  return EXIT_SUCCESS;
}

File: CSharp/PolyData/ClosedSurface.cs
using System;

using Kitware.VTK;

namespace ActiViz.Examples {
   class Program {
      static void Main(string[] args) {
         ClosedSurface();
      }


      private static void ClosedSurface() {
         // Create a sphere
         vtkSphereSource sphereSource = vtkSphereSource.New();
         sphereSource.Update();

         vtkFeatureEdges featureEdges = vtkFeatureEdges.New();
         featureEdges.FeatureEdgesOff();
         featureEdges.BoundaryEdgesOn();
         featureEdges.NonManifoldEdgesOn();
         featureEdges.SetInputConnection(sphereSource.GetOutputPort());
         featureEdges.Update();

         int numberOfOpenEdges = featureEdges.GetOutput().GetNumberOfCells();

         if(numberOfOpenEdges > 0) {
            Console.WriteLine("Surface is not closed");
         }
         else {
            Console.WriteLine("Surface is closed");
         }
         // nothing to show graphically
         Console.WriteLine("\nPress any key to continue...");
         Console.ReadKey();
      }
   }
}
